model/trigger/generate.go

```diff
@@ -22,28 +22,31 @@ type notificationGenerator struct {
 	jiraComment      *string
 	githubStatusAPI  *message.GithubStatus
 	slack            *notification.SlackPayload
+	// filterFunc takes an event and extra subscriber data and returns
+	// true if the notification should still be sent, false if not
+	filterFunc func(*event.EventLogEntry, map[string]string) bool
```

Comment:
This functionality is the responsibility of the trigger, so it doesn't belong here. I was imagining passing the TriggerData to the trigger functions. 

Furthermore, this late filtering has a problem in that the notifications produced will be vague. Ex: "version exceeded allotted time" vs. "version exceeded 10 seconds". Ex 2: "version duration exceeded threshold compared to last green" vs. "version duration deviated by more than 10% of the last green".